pkg/apis/build/v1alpha1/image_builds.go

```diff
@@ -17,18 +16,35 @@ const (
 	ImageLabel           = "image.kpack.io/image"
 	ImageGenerationLabel = "image.kpack.io/imageGeneration"
 
-	BuildReasonAnnotation = "image.kpack.io/reason"
-	BuildNeededAnnotation = "image.kpack.io/additionalBuildNeeded"
+	BuildReasonAnnotation  = "image.kpack.io/reason"
+	BuildChangesAnnotation = "image.kpack.io/buildChanges"
+	BuildNeededAnnotation  = "image.kpack.io/additionalBuildNeeded"
 
 	BuildReasonConfig    = "CONFIG"
 	BuildReasonCommit    = "COMMIT"
 	BuildReasonBuildpack = "BUILDPACK"
 	BuildReasonStack     = "STACK"
 	BuildReasonTrigger   = "TRIGGER"
+	BuildReasonSortIndex = "TRIGGER,COMMIT,CONFIG,BUILDPACK,STACK"
```

Comment:
can this be a map?

Reply:
confused as to how changing this to a map would help, Tyler? 😕 
It is currently used [here](https://github.com/pivotal/kpack/blob/9fcfdc89ea5a60ffde2506cf06ba1bc8e8cf749b/pkg/buildchange/change_processor.go#L72-L86) for sorting purpose only

Reply:
```
BuildReasonSortIndex := map[string]int{"TRIGGER":0,"COMMIT":1,"CONFIG":2,"BUILDPACK":3,"STACK":4}
sort.SliceStable(reasons, func(i, j int) bool {
		return BuildReasonSortIndexMap[v1alpha1.BuildReasonSortIndex][reasons[i]] <
			BuildReasonSortIndexMap[v1alpha1.BuildReasonSortIndex][reasons[i]])
	})
```

Reply:
@tylerphelan thanks for clarifying what you mean. Honestly I don't see considerable pros/cons between the 2 approaches, given that the BuildReasonSortIndex string is very small. I actually prefer the current implementation because the SortIndex is declared right next to the list of BuildReasons (and is a form of documentation)

https://github.com/pivotal/kpack/blob/072cf212feea355d2ec01d56a6bae4b13a6cf7dd/pkg/apis/build/v1alpha1/image_builds.go#L23-L28

the way they are used now
https://github.com/pivotal/kpack/blob/072cf212feea355d2ec01d56a6bae4b13a6cf7dd/pkg/buildchange/change_processor.go#L77-L111

Reply:
@tylerphelan please let me know if you feel strongly about this. I am open to discussion.

If not, would like resolve this conversation.

Reply:
`BuildReasonSortIndex` is sufficiently small so performance shouldn't really matter